### DISCUSSION TITLE
Post processing in ICEPAK using pyaedt

### DISCUSSION THREAD

**Discussion by pruthvi1991** [ansys/pyaedt#4899] [Q&A] [answered]
Hi, pru here using pyaedt for automating my icepak simulations. I have setup a conduction only simulation with several heat sources. The heat sources are assigned to objects inside a larger object i.e the heat source object resides within a much larger 'substrate' object. I would like to extract the following information using the postprocessing module:

1. min,avg,max temperature inside each of the heat source objects
2. temperature trace along a polyline (already defined)

I would like to extract this data and plot it using matplotlib. I read the documentation however could not figure out how to get this info.

Help appreciated!

**Comment by lorenzovecchietti** [comment, marked as answer]
I'm pasting here some code used in another example, but that should suit your request, please adapt to your needs. Let me know if you need anything else.

```
report = ipk.post.create_report(
    expressions=["Temperature", "Speed"],
    context=line.name,
    primary_sweep_variable="Distance",
    report_category="Fields",
    polyline_points=500,
)
report_data = report.get_solution_data()
distance = [k[0] for k, _ in report_data.full_matrix_mag_phase[0]["Temperature"].items()]
temperature = [v for _, v in report_data.full_matrix_mag_phase[0]["Temperature"].items()]
speed = [v for _, v in report_data.full_matrix_mag_phase[0]["Speed"].items()]

# Plot the data

fig, ax = plt.subplots(1, 1)
sc = ax.scatter(distance, speed, c=temperature)
ax.grid()
ax.set_xlabel("Distance [mm]")
ax.set_ylabel("Speed [m/s]")
cbar = fig.colorbar(sc)
cbar.set_label("Temperature [cel]")
```

**Reply by pruthvi1991** [reply]
Hi, thanks for sharing your example. Is this from the documentation? I tried it and it works. I'm going to modify it more for my needs.

**Reply by lorenzovecchietti** [reply]
Hi, it is from a new example that will be published soon.
